### PR TITLE
Fix for rr graph loading by VPR

### PR DESCRIPTION
**Pull request by mkurc-ant**
I've noticed that in the modified VPR version used by OpenFPGA segment ids are not set when a rr graph is loaded.

In this PR I've fixed that by creating segments in the `RRGraph` class and assigning them to nodes read from a rr graph XML file. I've also moved the essential check of the loaded rr graph after segment processing.

### CODE REVIEW

**Comment by mkurc-ant** [comment]
@tangxifan Can you take a look at this PR please?

File: vpr/src/device/rr_graph_obj.cpp [hunk 1]
```diff
@@ -449,6 +449,7 @@ void RRGraph::print_node(const RRNodeId& node) const {
     VTR_LOG("Node ptc: %d\n", node_ptc_num(node));
     VTR_LOG("Node num in_edges: %d\n", node_in_edges(node).size());
     VTR_LOG("Node num out_edges: %d\n", node_out_edges(node).size());
+    VTR_LOG("Node segment id: %d\n", node_segment(node));
```
Comment (tangxifan) [inline vpr/src/device/rr_graph_obj.cpp:452]: Not every node has a valid segment id. I suggest to output segment id only when the node is CHANX or CHANY

**Review by tangxifan — CHANGES_REQUESTED** [review]

**Comment by tangxifan** [comment]
> @tangxifan Can you take a look at this PR please?

Looks good to me. Just some minor comments and we can merge it. Thanks for the help!

File: vpr/src/device/rr_graph_obj.cpp (hunk 1 above)
Comment (mkurc-ant) [reply vpr/src/device/rr_graph_obj.cpp]: Ok, I've fixed that.

**Review by tangxifan — APPROVED** [review]

**Pull request merged**